src/traversals/bfs.jl

```diff
@@ -189,12 +189,15 @@ function is_bipartite(g::SimpleGraph, s::Int)
     return visitor.is_bipartite
 end
 
-function is_bipartite(g::SimpleGraph)
-    nvg = nv(g)
-    for v in vertices(g)
-        if !is_bipartite(g, v)
-            return false
+if VERSION >= v"0.4.0-dev+6371"
+    is_bipartite(g::SimpleGraph) = all([is_bipartite(g,v) for v in vertices(g)])
```

Comment:
Here's what we have:

```
    is_bipartite1(g::SimpleGraph) = all([is_bipartite(g,v) for v in vertices(g)])
    is_bipartite2(g::SimpleGraph) = all(x->is_bipartite(g,x), vertices(g))
    function is_bipartite(g::SimpleGraph)
        for v in vertices(g)
            if !is_bipartite(g, v)
                return false
            end
        end
        return true
    end
```

with results for a (10k, 100k) random digraph:

```
julia> @time LightGraphs.is_bipartite(g)
  31.225 microseconds (21 allocations: 94832 bytes)
false

julia> @time LightGraphs.is_bipartite1(g)
 388.163 milliseconds (170 k allocations: 910 MB, 29.60% gc time)
false

julia> @time LightGraphs.is_bipartite2(g)
  33.075 microseconds (25 allocations: 94944 bytes)
false
```

So it looks like `is_bipartite2()` is equivalent to the original, and much simpler. I'd rather rely on the Base-native short-circuiting behavior. Thoughts?